components/suggest/src/rs.rs

```diff
@@ -93,90 +103,85 @@ impl RemoteSettingsClient {
-            "fakespot-suggestions" => &self.fakespot_client,
+    fn client_for_record(&self, record: &Record) -> &remote_settings::Client {
+        match record.payload {
+            SuggestRecord::Fakespot => &self.fakespot_client,
             _ => &self.quicksuggest_client,
         }
     }
 }
 
 impl Client for RemoteSettingsClient {
-    fn get_records(&self, request: RecordRequest) -> Result<Vec<Record>> {
-        let client = self.client_for_record_type(request.record_type.as_str());
-        let options = request.into();
-        client
-            .get_records_with_options(&options)?
-            .records
-            .into_iter()
-            .map(|record| {
-                let attachment_data = record
-                    .attachment
-                    .as_ref()
-                    .map(|a| client.get_attachment(&a.location))
-                    .transpose()?;
-                Ok(Record::new(record, attachment_data))
-            })
-            .collect()
+    fn get_records(&self, db: &SuggestDb, fetch_fakespot_records: bool) -> Result<Vec<Record>> {
+        let records = get_records_for_collection(&self.quicksuggest_client, db, "quicksuggest")?;
+        if !fetch_fakespot_records {
+            Ok(records.collect())
+        } else {
+            let fakespot_records =
+                get_records_for_collection(&self.fakespot_client, db, "fakespot-suggest-products")?;
+            Ok(records.chain(fakespot_records).collect())
+        }
     }
-}
 
-#[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
-pub struct RecordRequest {
-    pub record_type: String,
-    pub last_modified: Option<u64>,
+    fn download_attachment(&self, record: &Record) -> Result<Vec<u8>> {
+        match &record.attachment {
+            Some(a) => Ok(self.client_for_record(record).get_attachment(&a.location)?),
+            None => Err(Error::MissingAttachment(record.id.to_string())),
+        }
+    }
 }
 
-impl From<RecordRequest> for GetItemsOptions {
-    fn from(value: RecordRequest) -> Self {
-        let mut options = GetItemsOptions::new();
-
-        // Remote Settings returns records in descending modification order
-        // (newest first), but we want them in ascending order (oldest first),
-        // so that we can eventually resume downloading where we left off.
-        options.sort("last_modified", SortOrder::Ascending);
-
-        options.filter_eq("type", value.record_type);
-
-        if let Some(last_modified) = value.last_modified {
-            options.filter_gt("last_modified", last_modified.to_string());
-        }
-        options
+fn get_records_for_collection(
+    client: &remote_settings::Client,
+    db: &SuggestDb,
+    collection_name: &'static str,
+) -> Result<impl Iterator<Item = Record>> {
+    // For now, handle the cache manually.  Once 6328 is merged, we should be able to delegate
+    // this to remote_settings.
+    let cache = db.read(|dao| dao.read_cached_rs_data(collection_name))?;
```

Comment:
As noted earlier, loading the cache over here feels a bit strange to me. Imagine if we want to handle cache failures later, that would leak more cache handling details into this module. Maybe we can move that to another place and and just consumer the cache data here.